etstool.py

```diff
@@ -264,20 +266,55 @@ def install(runtime, toolkit, environment, editable, source):
 
     click.echo("Creating environment '{environment}'".format(**parameters))
     execute(commands, parameters)
+    click.echo('Done install')
 
-    if source:
-        cmd_fmt = "edm plumbing remove-package --environment {environment} --force "
-        commands = [cmd_fmt+dependency for dependency in source_dependencies.keys()]
-        execute(commands, parameters)
-        source_pkgs = source_dependencies.values()
-        commands = [
-            "python -m pip install {pkg} --no-deps".format(pkg=pkg)
-            for pkg in source_pkgs
-        ]
-        commands = ["edm run -e {environment} -- " + command for command in commands]
-        execute(commands, parameters)
 
-    click.echo('Done install')
+@cli.command()
+@click.option('--runtime', default=DEFAULT_RUNTIME)
+@click.option('--toolkit', default=DEFAULT_TOOLKIT)
+@click.option('--environment', default=None)
+def install_from_source(runtime, toolkit, environment):
+    """ Install dependencies from source. """
+    parameters = get_parameters(runtime, toolkit, environment)
+
+    # Create the empty devenv
+    commands = ["edm environments create {environment} --force --version={runtime}"]
+    execute(commands, parameters)
+
+    install_additional_dependencies = [
+        "edm run -e {environment} -- pip install {additional_dependencies}",
+    ]
+    parameters["additional_dependencies"] = " ".join(ci_dependencies | {"sphinx", "enthought-sphinx-theme"})
+    execute(install_additional_dependencies, parameters)
+
+    # Install the traitsui package and its dependencies
+    extras = {"editors", "test"}
```

Comment:
Since we are not going to test the examples in the cron job (hence skipping the examples dependencies here), we need to be able to skip the following in the `test` command (see line 361 in this file, which I could not comment directly):
```
"edm run -e {environment} -- python -W default -m unittest discover -v {integrationtests}"
```
Which means... we might need yet another command / line in `.travis.yml` for the test command to be different from per-PR build.  Maybe it is good to split out the command for running the integration tests anyway.
GitHub actions should make these setups cleaner.